apm-agent-plugins/apm-dubbo-plugin/src/main/java/co/elastic/apm/agent/dubbo/helper/DubboTraceHelper.java

```diff
@@ -0,0 +1,162 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2020 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.dubbo.helper;
+
+import co.elastic.apm.agent.bci.VisibleForAdvice;
+import co.elastic.apm.agent.configuration.CoreConfiguration;
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.impl.context.Destination;
+import co.elastic.apm.agent.impl.transaction.Span;
+import co.elastic.apm.agent.impl.transaction.TraceContextHolder;
+import co.elastic.apm.agent.impl.transaction.Transaction;
+
+import java.net.InetSocketAddress;
+import java.net.URL;
+import java.security.CodeSource;
+import java.security.ProtectionDomain;
+
+@VisibleForAdvice
+public class DubboTraceHelper {
+
+    @VisibleForAdvice
+    public static ElasticApmTracer tracer;
+
+    private static String EXTERNAL_TYPE = "external";
+
+    private static String DUBBO_SUBTYPE = "dubbo";
+
+    public static final String PROVIDER_SERVICE_NAME_KEY = "elastic-apm-dubbo-provider";
+
+    public static void init(ElasticApmTracer tracer) {
+        DubboTraceHelper.tracer = tracer;
+    }
+
+    @VisibleForAdvice
+    public static String buildDubboRequestName(DubboApiInfo apiInfo) {
```

Comment:
This allocates lots of objects on each invocation. In order to reduce allocations, could we go for a more simple span name to make it completely allocation free like this:

```java
span.appendToName(apiInfo.getApiClass().getName())
    .appendToName("#")
    .appendToName(apiInfo.getMethodName());
```

Reply:
ok i will reduce the allocations